src/core/src/pass/sdpa_to_paged_attention.cpp

```diff
@@ -100,42 +101,65 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
                                                   max_context_len->output(0));
     manager.register_pass<PrevSequenceLengthPattern>(prev_max_seq_len, batch_dim);
     manager.register_pass<TotalSequenceLengthPattern>(max_context_len);
-
     manager.register_pass<PositionIDsReplacer>(unsqueezed_position_ids->output(0));
-
     manager.run_passes(model);
 
+    {
+        // Remove all Assigns aggressively, the path from the kv-cache concat to Assign can be complicated,
+        // but there is no reason to track it and reject part of the Assigns, because the model will remain
+        // in incorrect form anyway.
+        auto sinks = model->get_sinks();
+
+        for (auto& sink : sinks) {
+            model->remove_sink(sink);
+        }
+    }
+
+    for (auto& result : results_to_remove) {
```

Comment:
I think it's better to move it {} block above just in case

Reply:
Alright, lesson learned :)